### PR TITLE
test import without ssl module

### PR DESCRIPTION
**Pull request by t-8ch**
duplicates #586

### CODE REVIEW

**Comment by vbraun** [comment]
The problem is that urllib3 has already been imported by the time that the test is run, so "import urllib3" uses the cache. In fact, the Travis tests should fail because of #584 but passes.

I tried imp.reload / unloading urllib3* but couldn't get that to work properly. After reloading urllib3 you end up with a mix of the cached previous import and the new import somewhere in mock leading to test failures.

**Comment by t-8ch** [comment]
@vbraun The new version should be better

**Comment by t-8ch** [comment]
Now it breaks coverage...

File: test/test_no_ssl.py [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+            return self
+        return None
+
+    def load_module(self, fullname):
+        raise ImportError('import of {0} is blocked'.format(fullname))
+
+
+class ModuleStash(object):
+    """
+    Stashes avail previously imported modules
```
Comment (vbraun) [inline test/test_no_ssl.py]: Stashes away?

Comment (t-8ch) [inline test/test_no_ssl.py]: Yes, of course. No idea what my brain did there

File: test/test_no_ssl.py [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+    def stash(self):
+        self._data[self.namespace] = self.modules.pop(self.namespace, None)
+
+        for module in list(self.modules.keys()):
+            if module.startswith(self.namespace + '.'):
+                self._data[module] = self.modules.pop(module)
+
+    def pop(self):
+        for module in list(self.modules.keys()):
```
Comment (vbraun) [inline test/test_no_ssl.py]: Do you really need this loop? the sys.modules.update at the end overwrites them anyways. Also, you don't pop urllib3 (without dot).

**Comment by vbraun** [comment]
The approach looks good, if it works then thats great!

File: test/test_no_ssl.py (hunk 2 above)
Comment (t-8ch) [inline test/test_no_ssl.py]: I'd rather be more explicit. Maybe there are different modules imported while the stash was active.

**Comment by shazow** [comment]
Ah yes, this is a bit better!

**Comment by shazow** [comment]
... Except that tests fail, of course. :P

**Comment by t-8ch** [comment]
@shazow Here :-)
You may want to revert db95 onnce merged, to make the tests active

**Comment by shazow** [comment]
@t-8ch Sorry if I'm being dense, but why is db95ad4 required now but not later? :/

**Comment by t-8ch** [comment]
@shazow it is only there to show, that the test works. In reality it is failing currently

**Comment by shazow** [comment]
Then perhaps this PR should be on the repo that is fixing the problem, or we should hold off until that is merged?

**Comment by t-8ch** [comment]
Yeah. But to be pedantic: we are currently broken :sadface:

**Comment by shazow** [comment]
We're not broken, we're just differently-fixed.

**Comment by t-8ch** [comment]
@shazow @vbraun Care to review again?

**Comment by vbraun** [comment]
Sweet, you fixed it already! +1 from me

**Comment by shazow** [comment]
You're so eli7e.

**Pull request merged**